File: SQL/Expressions/SQLFieldAggregateExpression.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLFieldAggregateExpression
        Inherits SQLFieldExpression

        Private peAggregateFunction As AggregateFunction = AggregateFunction.None

        Public Sub New()


        End Sub

        Public Sub New(ByVal strFieldName As String)

            MyBase.New(strFieldName)

        End Sub

        Public Sub New(ByVal objTable As SQLSelectTable, ByVal strFieldName As String)

            MyBase.New(objTable, strFieldName)

        End Sub

        Public Sub New(ByVal objTable As SQLSelectTable, ByVal strFieldName As String, ByVal eAggregate As AggregateFunction)

            MyBase.New(objTable, strFieldName)
            Me.AggregateFunction = eAggregate

        End Sub

        Public Property AggregateFunction() As SQL.AggregateFunction
            Get

                Return peAggregateFunction

            End Get

            Set(ByVal Value As SQL.AggregateFunction)

                peAggregateFunction = Value

            End Set
        End Property

        Friend Overrides Function SQL(ByVal eConnectionType As Database.ConnectionType) As String

            If Me.AggregateFunction = AggregateFunction.None Then
                Throw New InvalidOperationException("AggregateFunction unspecified for " & Me.GetType.Name)
            End If

            Dim strFieldName As String = String.Empty

            If Me.AggregateFunction = AggregateFunction.Count Then
                strFieldName = "*"
            Else
                strFieldName = MyBase.SQL(eConnectionType)
            End If

            Return SQLConvertAggregate(Me.AggregateFunction) & "(" & strFieldName & ")"

        End Function

    End Class

End Namespace

File: SQL/Expressions/SQLFieldAggregateExpression.cs
// _________________________________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// 
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//	    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// _________________________________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLFieldAggregateExpression : SQLFieldExpression
	{
		private AggregateFunction peAggregateFunction = AggregateFunction.None;
			
		public SQLFieldAggregateExpression()
		{
		}
			
		public SQLFieldAggregateExpression(string strFieldName) 
            : base(strFieldName)
		{
		}
			
		public SQLFieldAggregateExpression(SQLSelectTable objTable, string strFieldName) 
            : base(objTable, strFieldName)
		{
		}
			
		public SQLFieldAggregateExpression(SQLSelectTable objTable, string strFieldName, AggregateFunction eAggregate) 
            : base(objTable, strFieldName)
		{
			this.AggregateFunction = eAggregate;
		}
			
		public SQL.AggregateFunction AggregateFunction
		{
			get
			{
				return peAggregateFunction;
			}
				
			set
			{
				peAggregateFunction = value;
			}
		}
			
		internal override string SQL(Serializers.Serializer serializer)
		{
			return serializer.SerializeFieldAggregateExpression(this);
		}
	}
}
